auth/clients.go

```diff
@@ -2,56 +2,122 @@ package auth
 
 import (
 	"fmt"
-	"os"
+	"strings"
 
+	"github.com/codegangsta/cli"
+	"github.com/jrperritt/rack/util"
 	"github.com/rackspace/gophercloud"
 	"github.com/rackspace/gophercloud/rackspace"
 )
 
-// NewClient creates and returns a Rackspace client for the given service.
-func NewClient(t string) *gophercloud.ServiceClient {
-	var err error
-	ao, region, err := authMethod()
-	if err != nil {
-		fmt.Printf("Error retrieving authentication credentials: %s\n", err)
+// reauthFunc is what the ServiceClient uses to re-authenticate.
+func reauthFunc(pc *gophercloud.ProviderClient, ao gophercloud.AuthOptions) func() error {
+	return func() error {
+		return rackspace.AuthenticateV2(pc, ao)
 	}
+}
+
+// NewClient creates and returns a Rackspace client for the given service.
+func NewClient(c *cli.Context, t string) (*gophercloud.ServiceClient, error) {
+	// get the user's authentication credentials
+	ao, region := Credentials(c)
+	// upper-case the region
+	region = strings.ToUpper(region)
+	// allow Gophercloud to re-authenticate
+	ao.AllowReauth = true
+	// if the user didn't provide an auth URL, default to the Rackspace US endpoint
```

Comment:
Wondering if it's safe to assume a region. The identity endpoint is global now anyway right? It's services that we had to not default to.